decoder/src/elf2table/mod.rs

```diff
@@ -13,17 +13,37 @@ use std::{
     path::{Path, PathBuf},
 };
 
-use crate::{BitflagsKey, StringEntry, Table, TableEntry, Tag, DEFMT_VERSION};
+use crate::{BitflagsKey, Encoding, StringEntry, Table, TableEntry, Tag, DEFMT_VERSION};
 use anyhow::{anyhow, bail, ensure};
 use object::{Object, ObjectSection, ObjectSymbol};
 
 pub fn parse_impl(elf: &[u8], check_version: bool) -> Result<Option<Table>, anyhow::Error> {
     let elf = object::File::parse(elf)?;
     // first pass to extract the `_defmt_version`
     let mut version = None;
-    let is_defmt_version = |name: &str| {
-        name.starts_with("\"_defmt_version_ = ") || name.starts_with("_defmt_version_ = ")
+    let mut encoding = None;
+
+    // Note that we check for a quoted and unquoted version symbol, since LLD has a bug that
+    // makes it keep the quotes from the linker script.
+    let try_get_version = |name: &str| {
+        if name.starts_with("\"_defmt_version_ = ") || name.starts_with("_defmt_version_ = ") {
+            Some(
+                name.trim_start_matches("\"_defmt_version_ = ")
+                    .trim_start_matches("_defmt_version_ = ")
+                    .trim_end_matches('"')
+                    .to_string(),
+            )
+        } else {
+            None
+        }
+    };
+
+    // No need to remove quotes for `_defmt_encoding_`, since it's
```

Comment:
Looks like we missed an incomplete sentence here

Reply:
Fixed in #568